dashboard/test/models/sections/omni_auth_section_test.rb

```diff
@@ -221,4 +221,24 @@ class OmniAuthSectionTest < ActiveSupport::TestCase
     assert_equal OmniAuthSection.column_for_attribute(:name).limit, section.name.length
     assert section.name.end_with?('...')
   end
+
+  test 'unarchive archived sections when imported' do
+    owner = create :teacher
+
+    section = OmniAuthSection.from_omniauth(
+      code: 'G-222222',
+      type: Section::LOGIN_TYPE_GOOGLE_CLASSROOM,
+      owner_id: owner.id,
+      students: [],
+      )
+    section.reload
+    section.hidden = true
+    section = OmniAuthSection.from_omniauth(
+      code: 'G-222222',
+      type: Section::LOGIN_TYPE_GOOGLE_CLASSROOM,
+      owner_id: owner.id,
+      students: [],
+        )
+    refute section.hidden
```

Comment:
Using the factory might be a better option to initially set up the section to test:
```suggestion
    section = create :section, user: owner, login_type: Section::LOGIN_TYPE_GOOGLE_CLASSROOM, hidden: true

    OmniAuthSection.from_omniauth(
      code: section.code,
      type: Section::LOGIN_TYPE_GOOGLE_CLASSROOM,
      owner_id: owner.id,
      students: [],
        )
    refute section.reload.hidden
```